packages/prelude/test/array-test.ts

```diff
@@ -124,12 +124,12 @@ describe('map', () => {
 describe('reduce', () => {
   it('reduce(f, x, []) === x', () => {
     const x = {}
-    assert.strictEqual(reduce(() => {}, x, []), x)
+    assert.strictEqual(reduce(() => ({}), x, []), x)
   })
 
   it('reduce(append, [], a) === a', () => {
     const a = [1, 2, 3]
-    const b = reduce((b, x) => append(x, b), [], a)
+    const b = reduce<number[], number>((b, x) => append(x, b), [], a)
```

Comment:
It's a bummer that TS can't infer this.  Oh well.

Reply:
Yeah, that's because TS infers `[]` as `never[]`